webapps/assembly/src/main/java/org/camunda/bpm/webapp/impl/security/auth/UserAuthenticationResource.java

```diff
@@ -102,20 +103,24 @@ public Response doLogin(
     ServletContext servletContext = request.getServletContext();
     Date cacheValidationTime = ServletContextUtil.getAuthCacheValidationTime(servletContext);
     if (cacheValidationTime != null) {
-      LOGGER.finest("Cache validation time: " + cacheValidationTime);
+      LOGGER.traceCacheValidationTime(cacheValidationTime);
       authentication.setCacheValidationTime(cacheValidationTime);
     }
 
     Set<String> authorizedApps = authentication.getAuthorizedApps();
 
     if (!authorizedApps.contains(appName)) {
+      LOGGER.infoWebappFailedLogin(username, "not authorized");
```

Comment:
Is the `isWebappsAuthenticationLoggingEnabled` check missing?

Reply:
Yes, thank you. I added it and also introduced a test for this scenario.